addons/account/models/account_move.py

```diff
@@ -5763,40 +5773,12 @@ def get_account_notification(moves, is_success: bool):
                 },
             ]
 
-        limit = job_count + 1
-        to_process = self.env['account.move'].search(
-            [('sending_data', '!=', False)],
-            limit=limit,
-        )
-        need_retrigger = len(to_process) > job_count
+        domain = [('sending_data', '!=', False)]
+        to_process = self.search(domain, limit=job_count).try_lock_for_update()
         if not to_process:
             return
-
-        to_process = to_process[:job_count]
-        if not self.env['res.company']._with_locked_records(to_process, allow_raising=False):
-            return
-
-        # Collect moves by res.partner that executed the Send & Print wizard, must be done before the _process
-        # that modify sending_data.
-        moves_by_partner = to_process.grouped(lambda m: m.sending_data['author_partner_id'])
-
-        self.env['account.move.send']._generate_and_send_invoices(
-            to_process,
-            from_cron=True,
-        )
-
-        for partner_id, partner_moves in moves_by_partner.items():
-            partner = self.env['res.partner'].browse(partner_id)
-            partner_moves_error = partner_moves.filtered(lambda m: m.sending_data and m.sending_data.get('error'))
-            if partner_moves_error:
-                partner._bus_send(*get_account_notification(partner_moves_error, False))
-            partner_moves_success = partner_moves - partner_moves_error
-            if partner_moves_success:
-                partner._bus_send(*get_account_notification(partner_moves_success, True))
-            partner_moves_error.sending_data = False
-
-        if need_retrigger:
-            self.env.ref('account.ir_cron_account_move_send')._trigger()
+        self.env['account.move.send']._generate_and_send_invoices(to_process, allow_raising=False, notification=get_account_notification)
```

Comment:
I don't like this modification. The extra `notification` parameter is confusing and to me, it's not part of `_generate_and_send_invoices` to manage the notification. Why do you even need all this diff? You are just refactoring the CRON after all. Just keep the code as before and just rewrite the CRON according the notify_progress without refactoring the business flows. Maybe some part could be improved but let this part as it was before for now

Reply:
I did this refactor because the wizard uses `sending_data` as a way to "return" values when called from the cron job. All of this, just to send a notification from the cron - and since I am refactoring the cron function, I changed it.